app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -772,7 +772,7 @@ public function joinTable($table, $bind, $fields = null, $cond = null, $joinType
         }
 
         // validate table
-        if (strpos($tableName, '/') !== false) {
+        if (!is_object($tableName) && strpos($tableName, '/') !== false) {
```

Comment:
I prefer a stricter check with `instanceof Zend_Db_Expr`

Reply:
actaully I don't understand why the original code was `if (strpos($tableName, '/') !== false)`, I wouldn't even check for anything, just that `$tableName` is a string

Reply:
@kiatng @fballiano like this? checking for string before making a string comparision seems to be the most resonable thing to do:

```suggestion
        if (is_string($tableName) && strpos($tableName, '/') !== false) {
```

Reply:
@fballiano If the table name contains a `/` it means it's a resource name like `core/url_rewrite`, so it should be converted to the raw table name `core_url_rewrite`.

Reply:
right!